### PR TITLE
Update Polish translation

### PR DESCRIPTION
**Pull request by michkry**

### CODE REVIEW

**Comment by oleg-derevenetz** [comment]
Hi @Arthusppp could you please take a look at this PR and share your opinion?

**Comment by oleg-derevenetz** [comment]
Hi @michkry @Arthusppp what is the status of this PR? Is it ready?

**Comment by Arthusppp** [comment]
> Hi @michkry @Arthusppp what is the status of this PR? Is it ready?

No, it is not ready. We will let you know, as soon as it is ready.

**Comment by oleg-derevenetz** [comment]
> No, it is not ready. We will let you know, as soon as it is ready.

OK, thanks! :)

File: files/lang/pl.po [hunk 1]
```diff
@@ -1132,89 +1130,103 @@ msgid ""
...
+msgstr "Armia zostanie przeniesiona do następnego scenariusza"
 
 msgid ""
 "The kingdom will have additional %{resource} at the start of the scenario."
-msgstr ""
+msgstr "Królestwo otrzyma dodatkowe %{resource} na starcie scenariusza."
 
 msgid "\"%{spell}\" spell will be carried over the scenario."
-msgstr ""
+msgstr "Zaklęcie \"%{spell}\" zostanie przeniesione do następnego scenariusza"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1165]: Corrected

File: files/lang/pl.po [hunk 2]
```diff
@@ -1132,89 +1130,103 @@ msgid ""
...
 
 msgid "%{hero} hero can be hired in the scenario."
-msgstr ""
+msgstr "Bohater %{hero} może być zwerbowany w tym scenariuszu"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1168]: Corrected

File: files/lang/pl.po [hunk 3]
```diff
@@ -1132,89 +1130,103 @@ msgid ""
...
 
 msgid "Defeated %{hero} hero will not appear in the subsequent scenarios."
-msgstr ""
+msgstr "Pokonany bohater %{hero} nie będzie dostępny w kolejnych scenariuszach"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1171]: Corrected

File: files/lang/pl.po [hunk 4]
```diff
@@ -1132,89 +1130,103 @@ msgid ""
...
 "The dwarves hail you, \"Any friend of Roland is a friend of ours.  You may "
 "pass.\""
 msgstr ""
+"Witają cię krasnoludy, \"Przyjaciel Rolanda jest naszym przyjacielem. Możesz "
+"przejść\""
 
 msgid ""
 "The ogres give you a grunt of recognition, \"Archibald's allies may pass.\""
 msgstr ""
+"Ogry chrząkają na znak uznania, \"Sprzymierzeńcy Archibalda mogą przejść\""
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1208]: Corrected

File: files/lang/pl.po [hunk 5]
```diff
@@ -1132,89 +1130,103 @@ msgid ""
...
 msgid "The dwarves recognize their allies and gladly join your forces."
 msgstr ""
-"Krasnoludy rozpoznają swoich sojuszników i chętnie przyłączają się do Twoich "
-"sił."
+"Krasnoludy rozpoznają w tobie sojusznika i chętnie przyłączają się do ciebie."
 
 msgid "The ogres recognize you as the Dwarfbane and lumber over to join you."
 msgstr ""
+"Ogry rozpoznają cię jako zmorę krasnoludów i powoli zmierzają ku tobie aby "
+"wesprzeć twoje siły."
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po]: I think we should stick to the translation from OG version. Changing Dwarfbane to "Pogromcę Krasnoludów". Today I've seen strange behaviour in the translation of rewards in Archibald campaign. I will look into it later on.

File: files/lang/pl.po [hunk 6]
```diff
@@ -1340,12 +1352,13 @@ msgstr "Pokaż bohaterów"
 msgid ""
 "The main hero will have \"%{artifact}\" artifact at the start of the "
 "scenario."
-msgstr ""
+msgstr "Główny bohater otrzyma artefakt \"%{artifact}\" na starcie scenariusza"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1355]: Corrected

File: files/lang/pl.po [hunk 7]
```diff
@@ -1132,89 +1130,103 @@ msgid ""
...
-msgstr ""
+msgstr " zawsze dołączy do twojej armii."
 
 msgid "\"%{artifact}\" artifact will be carried over the scenario."
 msgstr ""
+"Artefact \"%{artifact}\" zostanie przeniesiony do następnego scenariusza."
 
 msgid "The army will be carried over the scenario."
-msgstr ""
+msgstr "Armia zostanie przeniesiona do następnego scenariusza"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1158]: Corrected

File: files/lang/pl.po [hunk 8]
```diff
@@ -922,19 +922,19 @@ msgid ""
 msgstr ""
 
 msgid "Arrow's Flight"
-msgstr ""
+msgstr "Lot strzały"
 
 msgid "Island of Chaos"
-msgstr ""
+msgstr "Wyspa chaosu"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:928]: Corrected

File: files/lang/pl.po [hunk 9]
```diff
@@ -1068,10 +1066,10 @@ msgid "Power's End"
 msgstr "Pierścień mocy"
 
 msgid "The Eternal Scrolls"
-msgstr ""
+msgstr "Wieczny Zwój"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1069]: Corrected

File: files/lang/pl.po [hunk 10]
```diff
@@ -990,29 +990,27 @@ msgid ""
...
+msgstr "Miasta Graniczne"
 
 msgid "Conquer and Unify"
-msgstr ""
+msgstr "Podbój i Zjednoczenie"
 
-#, fuzzy
 msgid "Crazy Uncle Ivan"
-msgstr "Wuj Ivan"
+msgstr "Szalony Wuj Ivan"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po]: Changing all "wuj" words to capital first letter (Wuj)

File: files/lang/pl.po [hunk 11]
```diff
@@ -990,29 +990,27 @@ msgid ""
...
 
 msgid "The Wayward Son"
-msgstr ""
+msgstr "Nieobliczalny Syn"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1003]: I think we should stick to the OG. Changing the translation to "Knąbrny syn".

File: files/lang/pl.po [hunk 12]
```diff
@@ -990,29 +990,27 @@ msgid ""
...
 
 msgid "Ivory Gates"
-msgstr ""
+msgstr "Bramy z Kośći Słoniowej"
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po]: fheroes2 does not change the name of the castle, therefore changing the translation to "Ivory Gates"

File: files/lang/pl.po [hunk 13]
```diff
@@ -1176,8 +1176,8 @@ msgstr ""
 
 msgid "The ogres recognize you as the Dwarfbane and lumber over to join you."
 msgstr ""
-"Ogry rozpoznają cię jako zmorę krasnoludów i powoli zmierzają ku tobie aby "
-"wesprzeć twoje siły."
+"Ogry rozpoznają cię jako pogromce krasnoludów i powoli zmierzają ku tobie "
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1179]: Corrected the grammatical error in "pogromce". Also changing "Pogromcę Krasnoludów" so each of the word starts with capital letter. 

The 7th scenario of Archibald campaign is called "Władca smoków". In the future, we will have to keep in mind that the word "smoków" should start with capital letter in order to stick to the same capital letter pattern:

image

File: files/lang/pl.po [hunk 14]
```diff
@@ -1226,7 +1226,7 @@ msgstr ""
 "zwycięstwo będzie twoje.\""
 
 msgid "\"The Dwarfbane!!!!, run for your lives.\""
-msgstr "\"Zmora krasnoludów, ratuj się kto może!!!\""
+msgstr "\"Pogromca krasnoludów, ratuj się kto może!!!\""
```
[the remark this answers is not in the document]
Comment (michkry) [reply files/lang/pl.po:1229]: Corrected

File: files/lang/pl.po [hunk 15]
```diff
@@ -1132,89 +1130,104 @@ msgid ""
...
+"wykrzykuje, \"Kto jest na tyle odwarzny by dolaczyć do naszego "
+"nieustraszonego sprzymierzeńca?\" Członkowie grupy wiwatują i biegną do "
+"ciebie by dolączyć do twoich szeregów."
 
 msgid ""
 "The dwarves hail you, \"Any friend of Roland is a friend of ours.  You may "
 "pass.\""
 msgstr ""
+"Witają cię krasnoludy, \"Przyjaciel Rolanda jest naszym przyjacielem. Możesz "
+"przejść\""
```
Comment (Arthusppp) [inline files/lang/pl.po:1204]: Dot is missing here.

Comment (michkry) [reply files/lang/pl.po:1204]: Corrected

**Review by Arthusppp — APPROVED** [review]

**Comment by Arthusppp** [comment]
@oleg-derevenetz @ihhub this PR is ready for your review. 

@michkry  Many thanks for this! Great job!

**Comment by oleg-derevenetz** [comment]
@michkry Thank you very much and welcome to the project contributors!
@Arthusppp thank you for your help with the review!

**Pull request merged**